src/apps/ubeswap/plugin.ts

```diff
@@ -205,8 +205,10 @@ async function getFarmPositionDefinitions(
           {
             address: farm.rewardsTokenAddress.toLowerCase(),
             network,
+            category: 'claimable',
           },
         ],
+        availableShortcutIds: ['claim-reward'],
```

Comment:
It seems to me, that each shortcut might want to send different information to the client.
For example, the `claim-reward` probably wants to indicate the rewarded token and amount. I'm not sure if adding the `category: 'claimable',` is the best abstraction to achieve this.
What would happen if the reward is in another token? Or what if the amounts are changing related to an external factor (like the date).

So, I think this list should be something more like:

```
availableShortcuts: [{
id: 'claim-reward',
params: {
 rewardedToken: '0x1234',
 rewardedAmount: 1
}],
```

But I haven't thought about this too much, so maybe your approach is considering something that I'm missing.

Reply:
Another option would be to send this info when fetching the shortcuts instead of the positions, so if there is a huge calculation that we need to do to show some shortcut data to the user, we are not blocking the positions with it.

Reply:
> It seems to me, that each shortcut might want to send different information to the client. For example, the `claim-reward` probably wants to indicate the rewarded token and amount. I'm not sure if adding the `category: 'claimable',` is the best abstraction to achieve this. What would happen if the reward is in another token? Or what if the amounts are changing related to an external factor (like the date).
> 
> So, I think this list should be something more like:
> 
> ```
> availableShortcuts: [{
> id: 'claim-reward',
> params: {
>  rewardedToken: '0x1234',
>  rewardedAmount: 1
> }],
> ```
> 
> But I haven't thought about this too much, so maybe your approach is considering something that I'm missing.

Yes this is why I added the `category: 'claimable'` to the tokens.
I explained this more in this comment I just made: https://github.com/valora-inc/hooks/pull/69#discussion_r1196255089

What you're suggesting would also work. I represented it a bit differently with other possible future use cases in mind.

Reply:
> Another option would be to send this info when fetching the shortcuts instead of the positions, so if there is a huge calculation that we need to do to show some shortcut data to the user, we are not blocking the positions with it.

I see, I think we're fine expecting fetching the positions is where all the hard work happen. All amounts should be returned by positions. And this is one of the important piece of data needed for shortcuts.

I wanted to avoid redoing requests / calculations to fetch shortcuts (definitions).
That's why I ended up "tagging" positions with very light shortcut specific info.
After all, the current shortcuts we need are related to positions. We could still extend that to things that aren't positions though.

I also tried to see what it would look like to have shortcuts use the output of positions. But then it needed to dig through the positions and figure out which part of the positions are the rewards. It didn't look right.